pkg/prowloader/prow.go

```diff
@@ -0,0 +1,292 @@
+	if err := json.Unmarshal(jobJSON[16:len(jobJSON)-1], &results); err != nil {
+		return nil, err
+	}
+	return results["items"], nil
+}
+
+func (pl *ProwLoader) prowJobToJobRun(pj prow.ProwJob) error {
+	releaseRegex := regexp.MustCompile("pull-ci-.*([0-9]+.[0-9]+)-.*")
+	matches := releaseRegex.FindStringSubmatch(pj.Spec.Job)
+	release := "master"
+	if len(matches) > 0 {
+		release = matches[1]
+	}
+
+	var result sippyprocessingv1.JobOverallResult
+	switch pj.Status.State {
+	case prow.PendingState:
+		// Skip for now, only store runs in a terminal state
+		return nil
+	case prow.SuccessState:
+		result = sippyprocessingv1.JobSucceeded
+	case prow.AbortedState:
+		result = sippyprocessingv1.JobNoResults
+	default:
+		result = sippyprocessingv1.JobTestFailure
+	}
+
+	id, err := strconv.ParseInt(pj.Status.BuildID, 0, 64)
+	if err != nil {
+		return nil
+	}
+
+	if _, ok := pl.prowJobCache[pj.Spec.Job]; !ok {
+		dbProwJob := models.ProwJob{
+			Name:     pj.Spec.Job,
+			Release:  release,
+			Variants: pl.variantManager.IdentifyVariants(pj.Spec.Job),
+		}
+		err := pl.dbc.DB.Clauses(clause.OnConflict{UpdateAll: true}).Create(&dbProwJob).Error
+		if err != nil {
+			return errors.Wrapf(err, "error loading prow job into db: %s", pj.Spec.Job)
+		}
+		pl.prowJobCache[pj.Spec.Job] = &dbProwJob
+	} else {
+		// Ensure the job is up to date, especially for variants.
+		dbProwJob := pl.prowJobCache[pj.Spec.Job]
+		newVariants := pl.variantManager.IdentifyVariants(pj.Spec.Job)
+		if !reflect.DeepEqual(newVariants, dbProwJob.Variants) {
+			dbProwJob.Variants = newVariants
+			//pl.dbc.DB.Save(&dbProwJob)
+		}
+	}
+
+	if _, ok := pl.prowJobRunCache[uint(id)]; !ok {
+		pjURL, err := url.Parse(pj.Status.URL)
+		if err != nil {
+			return err
+		}
+
+		parts := strings.Split(pjURL.Path, pl.bktName)
+		if len(parts) == 2 {
+			tests, failures, err := pl.prowJobRunTestsFromGCS(parts[1][1:])
+			if err != nil {
+				return err
+			}
+
+			pl.dbc.DB.Save(&models.ProwJobRun{
+				ProwJob:       *pl.prowJobCache[pj.Spec.Job],
+				ProwJobID:     uint(id),
+				URL:           pj.Status.URL,
+				Timestamp:     pj.Status.StartTime,
+				OverallResult: result,
+				Tests:         tests,
+				TestFailures:  failures,
+				Succeeded:     result == sippyprocessingv1.JobSucceeded,
+			})
+		}
+	}
+
+	return nil
+}
+
+func (pl *ProwLoader) findOrAddTest(name string) uint {
+	if id, ok := pl.prowJobRunTestCache[name]; !ok {
+		test := &models.Test{}
+		pl.dbc.DB.Where("name = ?", name).Find(&test)
+		if test.ID == 0 {
+			test.Name = name
+			pl.dbc.DB.Save(test)
+		}
+		pl.prowJobRunTestCache[name] = test.ID
+		return test.ID
+	} else {
+		return id
+	}
+}
+
+func (pl *ProwLoader) findOrAddSuite(name string) *uint {
+	if name == "" {
+		return nil
+	}
+
+	if id, ok := pl.suiteCache[name]; !ok {
+		suite := &models.Suite{}
+		pl.dbc.DB.Where("name = ?", name).Find(&suite)
```

Comment:
How many suites did you end up with in your db? Curious how many are out there.

Reply:
387, but there are a lot with random names like `cluster_events_c0cb8a41-2c0b-4146-a000-36d4f7bfbeb5`